### PR TITLE
Partial Typings Revert

### PR DESCRIPTION
**Pull request by jshcrowthe**
We have had several issues (e.g. #393, #392, #387, #388) related to the typings change that was made.

This change reverts the changes that were made to the top-level `firebase` NPM package. Packages depending on that should no longer be broken, but this change preserves the changes in the Semver major 0 `@firebase/*` packages.

Users who are on the old types consuming through the `firebase` package, should see issues resolved, while users who are on the experimental `@firebase/*` packages should not have to update their apps again.

### CODE REVIEW

**Comment by hribab** [comment]
👍

**Comment by c-goosen** [comment]
This would be great. Holding on 4.8.0 until a consensus is reached on this.

File: integration/firebase-typings/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "scripts": {
+    "test": "tsc index.ts --outDir dist"
+  },
+  "dependencies": {
+    "firebase": "4.8.1"
+  },
+  "devDependencies": {
+    "typescript": "^2.4.2"
+  }
+}
```
Comment (hiranya911) [inline integration/firebase-typings/package.json]: Add new line

**Comment by jshcrowthe** [comment]
This has been prereleased under the `firebase@next` tag as version `4.8.2-0`. Please feel free to give this a shot and let me know if it isn't working!

**Comment by jshcrowthe** [comment]
Additionally those watching any of these: #393, #392, #387, #388 should give this a try as well. Thanks all for your patience!

**Comment by c-goosen** [comment]
Thanks @jshcrowthe & @hiranya911 ! Looks like its working. Been using it in a ionic project since yesterday. Will keep an eye out for any issues.

**Pull request merged**